File: backend/src/main/java/com/app/vending/iot/entity/Product.java
package com.app.vending.iot.entity;

import com.app.vending.iot.common.enums.ProductStatus;
import lombok.*;
import lombok.experimental.FieldDefaults;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Data
@Builder(toBuilder = true)
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "products")
@FieldDefaults(level = AccessLevel.PRIVATE)
public class Product {

    @Id
    String id;
    String machineId;
    String name;
    Double price;
    String image;
    ProductStatus status;
}

File: frontend/win/vending/vending/model/Product.cs
using System.Text.Json.Serialization;

namespace vending.model
{
    public class Product
    {
        [JsonPropertyName("id")]
        public string Id { get; set; }

        [JsonPropertyName("name")]
        public string Name { get; set; }

        [JsonPropertyName("price")]
        public double Price { get; set; }

        [JsonPropertyName("image")]
        public string Image { get; set; }

        [JsonPropertyName("status")]
        public string Status { get; set; }

        [JsonPropertyName("quantity")]
        public int Quantity { get; set; }
    }
}